syntax = "proto3";

service KeyValueService {
  rpc Set(SetRequest) returns (SetReply) {}
  rpc Get(GetRequest) returns (GetReply) {}
  rpc Merge(MergeRequest) returns (MergeReply) {}
}

message SetRequest {
  string key = 1;
  int64 value = 2;
}

message SetReply {
  bool success = 1;
  }

message GetRequest {
  string key = 1;
  }

message GetReply {
  int64 value = 1;
  }

message MergeRequest {
  repeated string key = 1;
  repeated int64 value = 2;
}

message MergeReply {
  bool success = 1;
}
